### ISSUE TITLE
[FEATURE] Control randomness of sample methods (optional seed)

### ISSUE THREAD

**Issue by han-ol** [stefanradev93/BayesFlow#666]
**Is your feature request related to a problem? Please describe.**
I want to persist a dictionary of samples to disk in an analysis pipeline for later use with reproducible randomness (fixed seed).
The main benefit I would get from this is that I can later check whether a deserialized approximator has changed: If samples are identical to the what was saved to disk, the approximator has not changed. If sample is stochastic, you need to use statistical tests to check if distributions are identical.

**Describe the solution you'd like**
Provide a public API for specifying a seed.
```python
workflow/approximator/inference_network/scoring_rule.sample(..., seed: keras.random.SeedGenerator | int | None = None)
```
These get passed on to internal stochastic components (like `DiagonalNormal`, used as `inference_network.base_distribution` AND a stochastic `_inverse` method). If seed=None (default) we recover the current behavior: components rely on their own `self.seed_generator` (https://keras.io/api/random/seed_generator/). Passing an integer seed to the sample method should create a locally scoped new seed generator that is shared by all of the components during the sample call.

**Describe alternatives you've considered**
If we do not provide public API, users can check the source code and rely on internals to get reproducible samples e.g. kind of like this 

```python
def _reset_model_seed_generators(model, seed: int) -> None:                                                                                                                 
    rng = np.random.default_rng(seed)                                                                                                                                       
    for layer in model._flatten_layers(include_self=True):                                                                                                                  
        for attr in vars(layer).values():                                                                                                                                   
            if isinstance(attr, keras.random.SeedGenerator):                                                                                                                
                derived = int(rng.integers(0, 2**31))                                                                                                                       
                attr.state.assign([derived, 0])
```

**Comment by han-ol** [comment]
The issue is resolved by #668.
